pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -705,7 +705,9 @@ func (s *DiscoveryServer) loadAssignmentsForClusterIsolated(proxy *model.Proxy,
 	}
 
 	// The service was never updated - do the full update
+	s.mutex.RLock()
```

Comment:
It looks like EndpointShardsByService has a method to Lock/Unlock why not to use it?`s.EndpointShardsByService[string(hostname)].mutex.Lock()`

Reply:
@sbezverk It is to protect `EndpointShardsByService` here, not its field.